src/App.tsx

```diff
@@ -1,46 +1,35 @@
+import { Navigate, Route, Routes } from 'react-router-dom';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { NavBar } from './NavBar/NavBar';
+import { HomePage } from './HomePage/HomePage';
+import { PageNotFound } from './PageNotFound/PageNotFound';
+import { TabsPage } from './TabsPage/TabsPage';
+import { Tab } from './types/Tab';
 
-// const tabs = [
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
+const tabs: Tab[] = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
 
-const App = () => (
+const App: React.FC = () => (
   <>
-    {/* Also requires <html class="has-navbar-fixed-top"> */}
-    <nav className="navbar is-fixed-top has-background-light" data-cy="nav">
-      <div className="navbar-menu">
-        <div className="navbar-start">
-          <a href="/" className="navbar-item isActive">Home</a>
-          <a href="/tabs" className="navbar-item isActive">Tabs</a>
-        </div>
-      </div>
-    </nav>
+    <NavBar />
 
     <div className="section">
-      <h1 className="title">Home page</h1>
-      <h1 className="title">Tabs page</h1>
-      <h1 className="title">Page not found</h1>
+      <Routes>
+        <Route path="/" element={<HomePage />} />
+        <Route
+          path="/home"
+          element={<Navigate to="/" replace />}
+        />
+        <Route path="/tabs" element={<TabsPage tabs={tabs} />}>
+          <Route path=":tabId" element={<TabsPage tabs={tabs} />} />
+        </Route>
```

Comment:
I would suggest reviewing this article
https://reactrouter.com/docs/en/v6/getting-started/concepts#index-routes